### ISSUE TITLE
Adds extra '/v2/...' in url for certain repositories

### ISSUE THREAD

**Issue by dd757** [davedoesdev/dxf#29]
My use case needs to list all versions in an ACR. I usually works for all images but for google-containers/kube-apiserver-amd64 inside the registry it is throwing an exception.
```
Traceback (most recent call last):
  File "/mnt/c/Users/daman.bawa/Desktop/crumb/src/entities.py", line 149, in get_dest_versions
    return Image._get_tags(self.dest, self.dest_user, self.dest_pass)
  File "/mnt/c/Users/daman.bawa/Desktop/crumb/src/entities.py", line 163, in _get_tags
    return dxf.DXF(dest[:bound], dest[bound + 1 :], auth=auth).list_aliases()
  File "/mnt/c/Users/daman.bawa/Desktop/crumb/.v/lib/python3.8/site-packages/dxf/__init__.py", line 773, in list_aliases
    return it if iterate else list(it)
  File "/mnt/c/Users/daman.bawa/Desktop/crumb/.v/lib/python3.8/site-packages/dxf/__init__.py", line 153, in __iter__
    response = self._meth("get", self._path, **self._kwargs)
  File "/mnt/c/Users/daman.bawa/Desktop/crumb/.v/lib/python3.8/site-packages/dxf/__init__.py", line 407, in _request
    return super(DXF, self)._base_request(method, self._repo_path + path, **kwargs)
  File "/mnt/c/Users/daman.bawa/Desktop/crumb/.v/lib/python3.8/site-packages/dxf/__init__.py", line 248, in _base_request
    _raise_for_status(r)
  File "/mnt/c/Users/daman.bawa/Desktop/crumb/.v/lib/python3.8/site-packages/dxf/__init__.py", line 83, in _raise_for_status
    r.raise_for_status()
  File "/mnt/c/Users/daman.bawa/Desktop/crumb/.v/lib/python3.8/site-packages/requests/models.py", line 940, in raise_for_status
    raise HTTPError(http_error_msg, response=self)
requests.exceptions.HTTPError: 404 Client Error: Not Found for url: https://acrlabsred.azurecr.io/v2/google-containers/kube-apiserver-amd64/v2/google-containers/kube-apiserver-amd64/tags/list?last=v1.14.10&n=100&orderby=

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "src/main.py", line 165, in <module>
    command_main()  # pylint: disable=no-value-for-parameter
  File "/mnt/c/Users/daman.bawa/Desktop/crumb/.v/lib/python3.8/site-packages/click/core.py", line 764, in __call__
    return self.main(*args, **kwargs)
  File "/mnt/c/Users/daman.bawa/Desktop/crumb/.v/lib/python3.8/site-packages/click/core.py", line 717, in main
    rv = self.invoke(ctx)
  File "/mnt/c/Users/daman.bawa/Desktop/crumb/.v/lib/python3.8/site-packages/click/core.py", line 956, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/mnt/c/Users/daman.bawa/Desktop/crumb/.v/lib/python3.8/site-packages/click/core.py", line 555, in invoke
    return callback(*args, **kwargs)
  File "src/main.py", line 161, in command_main
    main(config_file, dest, dry_run)
  File "src/main.py", line 127, in main
    entity.get_dest_versions(), key=StrFavouredLooseVersion
  File "/mnt/c/Users/daman.bawa/Desktop/crumb/src/entities.py", line 152, in get_dest_versions
    raise EntityDoesNotExist
exceptions.EntityDoesNotExist
```

I traced it to inside PaginatingResponse.__iter__ in going through while loop self._path and self._kwargs change to 
tags/list
```
self._path = /v2/google-containers/kube-apiserver-amd64/tags/list?last=v1.14.10&n=100&orderby=
self._kwargs = {}
```

while in cases it works they stop as in first iteration of while 
```
self._path = tags/list
self._kwargs = {'params': {'n': None}}
```

**Comment by dd757** [comment]
On Further inspection, it seems like ACR paginates at 100 when you try to get a large list of tags (196 in this case). However, the next they provide has full path i.e. start as /v2/... which is joinned with `self._base_url` in when calling `DXFBase._base_request` from `PaginatingResponse.__iter__`. Just adding code to remove repo and/or v2 from path if present should do the trick.

**Comment by davedoesdev** [comment]
@dd757 thanks for investigating this. I'll take a look.

**Comment by davedoesdev** [comment]
@dd757 I think the bug might be in `_request` where I just concatenate `path`:

```javascript
    def _request(self, method, path, **kwargs):
        return super(DXF, self)._base_request(method,
                                              self._repo_path + path,
                                              **kwargs)
```

I think `_base_request` uses `path` correctly:

```javascript
        url = urlparse.urljoin(self._base_url, path)
```

**Comment by davedoesdev** [comment]
@dd757 could you try this change to see if it fixes the issue please?

```diff
diff --git a/dxf/__init__.py b/dxf/__init__.py
index 1555e0b..71021a5 100644
--- a/dxf/__init__.py
+++ b/dxf/__init__.py
@@ -376,9 +376,10 @@ class DXF(DXFBase):
         self._repo_path = (repo + '/') if repo else ''
 
     def _request(self, method, path, **kwargs):
-        return super(DXF, self)._base_request(method,
-                                              self._repo_path + path,
-                                              **kwargs)
+        return super(DXF, self)._base_request(
+            method,
+            urlparse.urljoin(self._repo_path, path),
+            **kwargs)
 
     def push_blob(self,
                   filename=None,
```

**Comment by dd757** [comment]
It is still giving same error.
From
```
def list_repos(self, batch_size=None, iterate=False):
        """
        List all repositories in the registry.

        :param batch_size: Number of repository names to ask the server for at a time.
        :type batch_size: int

        :param iterate: Whether to return iterator over the names or a list of all the names.
        :type iterate: bool

        :rtype: list or iterator of strings
        :returns: Repository names.
        """
        it = PaginatingResponse(
            self, "_base_request", "_catalog", "repositories", params={"n": batch_size}
        )
```
and then,
```
class PaginatingResponse(object):
    # pylint: disable=too-few-public-methods
    def __init__(self, dxf_obj, req_meth, path, header, **kwargs):
        self._meth = getattr(dxf_obj, req_meth)
        self._path = path
        self._header = header
        self._kwargs = kwargs

    def __iter__(self):
        while self._path:
            response = self._meth("get", self._path, **self._kwargs)
            self._kwargs = {}
            for v in response.json()[self._header] or []:
                yield v
            nxt = response.links.get("next")
            self._path = nxt["url"] if nxt else None
```
I think you are calling _base_request whic is in DXFBase. So it should not pass through 
For reference my call is following inside a function with parameters user and pass_,
```
def auth(dxf, response):
        dxf.authenticate(user, pass_, response=response)

bound = dest.find("/")

dxf.DXF(dest[:bound], dest[bound + 1 :], auth=auth).list_aliases()
```

**Comment by davedoesdev** [comment]
`list_aliases` passes `_request` as the method so it should pass through `_request`

**Comment by davedoesdev** [comment]
After the following line in `_base_request`:

```javascript
        url = urlparse.urljoin(self._base_url, path)
```

could you add this line so we can see the URL:

```
        print(self._base_url, path)
```

Thanks again

**Comment by dd757** [comment]
Yes it is passing through `_request`. Sorry about that, I am using two venvs so I got confused. It fixed the problem.

Thanks.

**Comment by davedoesdev** [comment]
Ah cool, I appreciate your help on the issue

**Comment by davedoesdev** [comment]
Fix is in version 7.5.5
